src/cwe_checker_lib/src/ghidra_pcode/tests.rs

```diff
@@ -5,43 +5,51 @@ use crate::{bitvec, def, expr, variable};
 
 fn mock_varnode(addressspace: &str, id: &str, size: u64) -> VarnodeSimple {
     VarnodeSimple {
-        addressspace: addressspace.to_string(),
+        address_space: addressspace.to_string(),
         id: id.to_string(),
         size,
     }
 }
 
-fn mock_pcode_op(
-    pcode_index: u64,
-    pcode_mnemonic: PcodeOperation,
+/// Simplified construction of pcode operation with `pcode_index: 1` and
+/// pcode_mnemonic: ExpressionType(INT_ADD).
+fn mock_pcode_op_add(
     input0: VarnodeSimple,
     input1: Option<VarnodeSimple>,
-    input2: Option<VarnodeSimple>,
     output: Option<VarnodeSimple>,
 ) -> PcodeOpSimple {
     PcodeOpSimple {
-        pcode_index,
-        pcode_mnemonic,
+        pcode_index: 1,
+        pcode_mnemonic: ExpressionType(INT_ADD),
         input0,
         input1,
-        input2,
+        input2: None,
         output,
     }
 }
-/// Simplified construction of pcode operation with `pcode_index: 1` and
-/// pcode_mnemonic: ExpressionType(INT_ADD).
-fn mock_pcode_op_add(
-    input0: VarnodeSimple,
-    input1: Option<VarnodeSimple>,
-    input2: Option<VarnodeSimple>,
-    output: Option<VarnodeSimple>,
-) -> PcodeOpSimple {
-    mock_pcode_op(1, ExpressionType(INT_ADD), input0, input1, input2, output)
-}
 
-impl<T> Term<T> {
-    fn set_tid_id(mut self, id: String) -> Term<T> {
-        self.tid.id = id;
+impl PcodeOpSimple {
+    fn set_mnemonic(&self, mnemonic: PcodeOperation) -> PcodeOpSimple {
+        PcodeOpSimple {
+            pcode_index: self.pcode_index,
+            pcode_mnemonic: mnemonic,
+            input0: self.input0.clone(),
+            input1: self.input1.clone(),
+            input2: self.input2.clone(),
+            output: self.output.clone(),
+        }
+    }
+    fn set_varnodes(
+        mut self,
```

Comment:
These two functions should again use the `with_x` naming scheme instead of the `set_x` naming scheme, since they use the builder pattern where one need to use the returned value after calling the function.